### PR TITLE
[msbuild] Unify MtouchDebug and MmpDebug into _Debug.

### PR DESCRIPTION
**Pull request by rolfbjarne**

### CODE REVIEW

File: msbuild/Xamarin.Shared/Xamarin.Shared.props [hunk 1]
```diff
@@ -96,6 +96,14 @@ Copyright (C) 2020 Microsoft. All rights reserved.
...
 		<TargetArchitectures Condition="'$(TargetArchitectures)' == '' And '$(_PlatformName)' != 'macOS' And '$(ComputedPlatform)' == 'iPhone'">ARMv7</TargetArchitectures>
 
+		<!-- Debug -->
+		<!-- Xamarin.Mac: use MmpDebug, but if that isn't set, use DebugSymbols -->
+		<_Debug Condition="'$(_PlatformName)' == 'macOS'">$(MmpDebug)</_Debug>
+		<_Debug Condition="'$(_PlatformName)' == 'macOS' And '$(_Debug)' == ''">$(DebugSymbols)</_Debug>
+		<!-- Xamarin.iOS: Use MtouchDebug -->
+		<_Debug Condition="'$(_PlatformName)' != 'macOS'">$(MtouchDebug)</_Debug>
+		<!-- The default is false for all platforms -->
+		<_Debug Condition="'$(_Debug)' == ''">false</_Debug>
```
[the remark this answers is not in the document]
Comment (rolfbjarne) [reply msbuild/Xamarin.Shared/Xamarin.Shared.props:106]: > I'm beginning to question the performance impact of those repetitive checks.

🤔 

> can't the have _personalized_ (platform-ized) versions of

Yes, we could precompile them, both for the current Xamarin.iOS/Mac packages as well as the new NuGets.

> we should measure performance before doing anything.

😁 

It's an interesting idea, so here are a script and some numbers: https://gist.github.com/rolfbjarne/00212e85b39d0162c9a916adce038549

TL;DR: doing 100k Condition checks adds 0,29s if the variable comparison succeeds, and 0,1 s if the variable comparison fails.

I think we can probably find other places where we get a better return on investment for our time doing performance work 🙂

**Comment by rolfbjarne** [comment]
> Isn't `_Debug` a bit generic? Just thinking that it is easier to clash with some other random nuget, maybe something like `_ XamarinDebug` or `_XamDebug`.

What about `_BundlerDebug`?

**Comment by dalexsoto** [comment]
@rolfbjarne yup `_BundlerDebug` sounds fine to me
